src/terminal/sys/unix.rs

```diff
@@ -53,6 +53,10 @@ pub(crate) fn window_size() -> io::Result<WindowSize> {
     };
 
     if wrap_with_result(unsafe { ioctl(fd, TIOCGWINSZ.into(), &mut size) }).is_ok() {
+        if size.ws_row == 0 && size.ws_col == 0 {
```

Comment:
Add a comment that tells the future reader why this is necessary (like the PR message)

Reply:
I added a comment.